src/Uno.UI/UI/Input/GestureRecognizer.Manipulation.cs

```diff
@@ -153,15 +153,60 @@ private Manipulation(GestureRecognizer recognizer, PointerPoint pointer1)
 				}
 
 				_isDraggingEnable = (_settings & GestureSettings.Drag) != 0;
-				_isTranslateXEnabled = (_settings & (GestureSettings.ManipulationTranslateX | GestureSettings.ManipulationTranslateRailsX)) != 0;
-				_isTranslateYEnabled = (_settings & (GestureSettings.ManipulationTranslateY | GestureSettings.ManipulationTranslateRailsY)) != 0;
+				_isTranslateXEnabled = (_settings & GestureSettings.ManipulationTranslateX) != 0;
+				_isTranslateYEnabled = (_settings & GestureSettings.ManipulationTranslateY) != 0;
 				_isRotateEnabled = (_settings & GestureSettings.ManipulationRotate) != 0;
 				_isScaleEnabled = (_settings & GestureSettings.ManipulationScale) != 0;
 
 				_recognizer.ManipulationConfigured?.Invoke(_recognizer, this);
 				StartDragTimer();
 			}
 
+			private bool _isTranslateDirectionDecided;
+
+			private void DecideTranslateDirection(Point oldPosition, Point newPosition, ref ManipulationDelta cumulative, ref ManipulationDelta delta, ref ManipulationVelocities velocities)
```

Comment:
I would also move that method at the end of the class, with other private methods that are only "helpers" to maintain the state, in order to keep the "core functionality / API" of the class at the top.

Reply:
Done